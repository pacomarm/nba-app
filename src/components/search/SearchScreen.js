import React from 'react'
import { players } from '../../data/players'
import { useForm } from '../../hooks/useForm';
import { PlayerCard } from '../players/PlayerCard';

export const SearchScreen = () => {
    const playersFiltered = players;
    
    const [values, handleInputChange, reset] = useForm({text: ''});

    const {text} = values;

    const handleSearch = (e) => {
        e.preventDefault();
        console.log(text);
    }

    return (
        <div>
            <h1>Search Screen</h1> 
            <hr />

            <div className="row">
                <div className="col-5">
                    <h4>Search Form</h4>
                    <hr/>
                    <form onSubmit={handleSearch}>
                        <input 
                            type="text" 
                            placeholder="Find a player"
                            className="form-control"
                            autoComplete="off"
                            name="text"
                            value={text}
                            onChange={handleInputChange}
                        />
                        <button 
                            className="mt-3 btn btn-block btn-outline-primary"
                            style={{width: "30%"}}
                            // onClick={handleReturn}
                        >
                            Search
                        </button>
                    </form>
               </div>
               <div className="col-7">
                    <h4>Results</h4>
                    <hr />
                    {
                        playersFiltered.map((player) => (
                            <PlayerCard
                                key={player.id}
                                {...player}
                            />
                        ))
                    }
               </div>
           </div>

        </div>
    )
}
