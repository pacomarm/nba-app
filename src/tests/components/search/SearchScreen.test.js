import React from 'react';
import { mount, configure } from "enzyme";
import { MemoryRouter, Redirect, Route } from 'react-router-dom';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { AuthContext } from '../../../auth/AuthContext';
import { types } from '../../../types/types';
import { SearchScreen } from '../../../components/search/SearchScreen';
configure({adapter: new Adapter()});


describe('tests Login', () => {

    const contextValue = {
        dispatch: jest.fn(),
    }
    
    
    test('should show correctly with default values', () => {
        
        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/search' ]}>
               <Route path="/search" component={SearchScreen}/>
            </MemoryRouter>
        );
        expect( wrapper).toMatchSnapshot();
        expect( wrapper.find('.alert-info').text().trim() ).toBe('Look for a player');

    });

    test('should show Lebron given q string', () => {
        
        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/search?q=Lebron' ]}>
               <Route path="/search" component={SearchScreen}/>
            </MemoryRouter>
        );
        expect( wrapper.find('input').prop('value') ).toBe('Lebron');
        expect( wrapper).toMatchSnapshot();
    });

    test('should show error if player is not found', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/search?q=Lebron121' ]}>
               <Route path="/search" component={SearchScreen}/>
            </MemoryRouter>
        );
        expect( wrapper.find('.alert-warning').text().trim() ).toBe('No player was found');
        expect( wrapper).toMatchSnapshot();
    });

    test('should call historys PUSH', () => {
        
        const history = {
            push: jest.fn(),
        }
        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/search?q=Lebron121' ]}>
                <Route 
                    path="/search" 
                    component={ () => <SearchScreen history={history}/> }

                />
            </MemoryRouter>
        );
        wrapper.find('input').simulate('change', {
            target: {
                name: 'text',
                value: 'lebron'
            }
        });

        wrapper.find('form').prop('onSubmit')({
            preventDefault(){}
        });
        expect( history.push ).toHaveBeenCalledWith('?q=lebron');

    })
    
    
    
    
})
