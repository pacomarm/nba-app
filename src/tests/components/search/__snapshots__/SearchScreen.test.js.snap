// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests Login should show Lebron given q string 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="text"
          onChange={[Function]}
          placeholder="Find a player"
          type="text"
          value="Lebron"
        />
        <button
          className="mt-3 btn btn-block btn-outline-primary"
          style={
            Object {
              "width": "30%",
            }
          }
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-6"
          >
            <img
              alt="LeBron James"
              className="card-img p-1"
              src="./assets/lakers-lbj.jpg"
            />
          </div>
          <div
            className="col-md-6"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                LeBron James
              </h5>
              <p
                className="card-text"
              >
                LeGoat
              </p>
              <p
                className="card-text"
              >
                Lebron #23
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  2003 Season
                </small>
              </p>
              <a
                href="/player/lakers-lbj"
                onClick={[Function]}
              >
                 more...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`tests Login should show correctly with default values 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="text"
          onChange={[Function]}
          placeholder="Find a player"
          type="text"
          value=""
        />
        <button
          className="mt-3 btn btn-block btn-outline-primary"
          style={
            Object {
              "width": "30%",
            }
          }
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Look for a player
      </div>
    </div>
  </div>
</div>
`;

exports[`tests Login should show error if player is not found 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="text"
          onChange={[Function]}
          placeholder="Find a player"
          type="text"
          value="Lebron121"
        />
        <button
          className="mt-3 btn btn-block btn-outline-primary"
          style={
            Object {
              "width": "30%",
            }
          }
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-warning"
      >
        No player was found
      </div>
    </div>
  </div>
</div>
`;
